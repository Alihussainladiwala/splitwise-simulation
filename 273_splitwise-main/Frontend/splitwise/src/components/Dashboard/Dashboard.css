.div-dashboard {
  height: 70vh;
}

.button-dashboard-groups {
  background-color: #ccc;
  border: 0px;
  color: #999;
}

.button-dashboard-recent-activity {
  color: #ff652f;
  font-weight: 800;
}

.button-dashboard-dashboard {
  color: #5bc5a7;
  font-weight: 800;
}

.dashboard-layout-row {
  width: 100vw;
  height: 90vh;
}

.links-dashboard-groups {
  color: #999;
  margin-left: 10px;
}

.total-balance {
  margin-left: 15px;
}

.tiles {
  padding: 0px !important;
  border: solid;
  border-bottom: 0px;
  border-color: gray;
}

.total-balance {
  padding-left: 0px !important;
  padding-right: 0px !important;
}

.you-owe {
  padding: 0px !important;
}

.you-are-owed {
  padding-left: 0px !important;
}

.nav-dashboard {
  background-color: #f8f9fa !important;
  padding: 0px !important;
}

.owe-owed {
  height: 50vh;
  padding: 0px !important;
}

.owe-column {
  border-right: solid;
  border-color: gray;
  padding-right: 0px !important;
}

.owed-column {
  padding-left: 0px !important;
}

.you-owe-title {
  height: 5vh;
  margin-left: 140px;
}
