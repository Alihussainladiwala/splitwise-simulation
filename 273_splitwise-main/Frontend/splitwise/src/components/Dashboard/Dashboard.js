/* eslint-disable react/jsx-filename-extension */
/* eslint-disable */
import React, { useEffect, useState } from 'react';
import {
  Nav,
  Row,
  Col,
  Button,
  Card,
  DropdownButton,
  ButtonGroup,
  Dropdown,
  Container,
  ListGroup,
  Modal,
  Form,
} from 'react-bootstrap';
import Axios from 'axios';
import './Dashboard.css';
import VerticalNav from '../VerticalNav/VerticalNav';
import cookie from 'react-cookies';
import { Redirect } from 'react-router-dom';
import Cookies from 'js-cookie';
import SideBar from '../SideBar/SideBar';
const queryString = require('query-string');
import 'numeral/locales/en-gb';
import endPointObj from '../../endPointUrl';
import { useDispatch, useSelector } from 'react-redux';

var numeral = require('numeral');

function Dashboard() {
  const [show, setShow] = useState(false);
  const [showOwed, setShowOwed] = useState(false);

  const email = useSelector((state) => state.login.username);

  const handleClose = () => {
    setAmount('money!');
    setShow(false);
    setShowOwed(false);
  };
  const handleShow = () => setShow(true);

  const handleShowOwed = () => setShowOwed(true);

  const [groups, setGroups] = useState([]);
  const [invites, setInvites] = useState([]);
  const [owe, setYouOwe] = useState([]);
  const [owed, setYouAreOwed] = useState([]);
  const [amount, setAmount] = useState('money!');
  const [reciever, setReciever] = useState('');
  const [totalBalance, setTotalBalance] = useState(0);
  const [youOwe, setYouOweTotal] = useState(0);
  const [youOwed, setYouOwedTotal] = useState(0);

  if (Cookies.get('name')) {
    console.log('loaded cookie');
  }

  const getGroups = () => {
    Axios.get(endPointObj.url + 'getGroups/' + email, {
      headers: {
        Authorization: 'Bearer ' + localStorage.getItem('token'),
      },
    })
      .then((response) => {
        // eslint-disable-next-line no-console
        console.log(response);
        setGroups(response.data);
      })
      .catch((e) => {
        console.log(e);
      });
  };

  useEffect(() => {
    // if (Cookies.get('userId')) {
    //   console.log('loaded cookie');
    // }
    numeral.reset();
    numeral.defaultFormat('$0,0.00');

    numeral.locale(sessionStorage.getItem('currency'));

    // const parsed = queryString.parse(location.search);
    // let email = parsed;
    getGroups();

    getInvites().then((result) => {
      setInvites(result);
    });

    fetchAmount();
  }, []);

  function fetchAmount() {
    getAmount(email).then((result) => {
      let youOwe = [];
      let youOwed = [];
      if (result == null || result == 'undefined' || result == '') {
        result = [];
      }
      result.map((person) => {
        if (person.amount > 0) {
          youOwed.push(person);
        }
        if (person.amount < 0) {
          person.amount = person.amount * -1;
          youOwe.push(person);
        }
      });

      let youOweAccum = 0;
      let youOwedAccum = 0;
      for (let i = 0; i < youOwe.length; i++) {
        youOweAccum = youOweAccum + youOwe[i].amount;
      }

      for (let i = 0; i < youOwed.length; i++) {
        youOwedAccum = youOwedAccum + youOwed[i].amount;
      }
      setYouOwe(youOwe);
      setYouAreOwed(youOwed);
      setYouOweTotal(youOweAccum);
      setYouOwedTotal(youOwedAccum);
      setTotalBalance(youOwedAccum - youOweAccum);
    });
  }

  const accept = (groupName, email) => {
    acceptInvite(groupName, email).then((result) => {
      getInvites(email).then((result) => {
        setInvites(result);
        getGroups(email);
      });
    });
  };

  function acceptInvite(groupName) {
    return new Promise((resolve, reject) => {
      Axios.post(
        endPointObj.url + 'inviteStatus',
        {
          status: true,
          groupName: groupName,
          email: email,
        },
        {
          headers: {
            Authorization: 'Bearer ' + localStorage.getItem('token'),
          },
        }
      )
        .then((response) => {
          // eslint-disable-next-line no-console
          console.log(response.data);
          resolve(response.data);
        })
        .catch((e) => {
          console.log(e);
        });
    });
  }

  function getAmountforOwe(e) {
    console.log(owe);
    let amountOwedList = owe.filter((list) => list.email == e.target.value);
    setAmount(amountOwedList[0].amount);
    setReciever(amountOwedList[0].email);
  }

  function getAmountforOwed(e) {
    console.log(e.target.value);
    console.log(owed);
    let amountOwedList = owed.filter((list) => list.email == e.target.value);
    setAmount(amountOwedList[0].amount);
    setReciever(amountOwedList[0].email);
  }

  function getInvites() {
    return new Promise((resolve, reject) => {
      Axios.get(endPointObj.url + 'invites/' + email, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
        .then((response) => {
          // eslint-disable-next-line no-console
          resolve(response.data);
        })
        .catch((e) => {
          console.log(e);
        });
    });
  }

  const settleUp = (sender, reciever, amount) => {
    return new Promise((resolve, reject) => {
      Axios.post(
        endPointObj.url + 'settleUp',
        {
          sender: sender,
          reciever: reciever,
          amount: amount,
        },
        {
          headers: {
            Authorization: 'Bearer ' + localStorage.getItem('token'),
          },
        }
      )
        .then((response) => {
          // eslint-disable-next-line no-console
          fetchAmount(email);
          setShow(false);
          setShowOwed(false);
          resolve(response.data);
        })
        .catch((e) => {
          console.log(e);
        });
    });
  };

  function getAmount(email) {
    return new Promise((resolve, reject) => {
      Axios.get(endPointObj.url + 'amount/' + email, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
        .then((response) => {
          // eslint-disable-next-line no-console
          console.log(response.data);
          resolve(response.data);
        })
        .catch((e) => {
          console.log(e);
        });
    });
  }

  return (
    <div className="div-dashboard">
      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Settle up</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form>
            <Form.Group controlId="exampleForm.ControlSelect1">
              <Form.Label>Pay</Form.Label>
              <Form.Control as="select" onChange={getAmountforOwe}>
                <option selected disabled hidden>
                  Choose here
                </option>
                {owe.map((a) => (
                  <option key={a.email} value={a.email}>
                    {a.name}
                  </option>
                ))}
              </Form.Control>
            </Form.Group>
            <Form.Group controlId="exampleForm.ControlInput1">
              <Form.Label>Amount</Form.Label>
              <Form.Control type="number" placeholder={numeral(amount).format()} disabled />
            </Form.Group>
          </Form>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Close
          </Button>
          <Button variant="primary" onClick={() => settleUp(email, reciever, amount)}>
            Settle up
          </Button>
        </Modal.Footer>
      </Modal>

      <Modal show={showOwed} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Settle up</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form>
            <Form.Group controlId="exampleForm.ControlSelect1">
              <Form.Label>Pay</Form.Label>
              <Form.Control as="select" onChange={getAmountforOwed}>
                <option selected disabled hidden>
                  Choose here
                </option>
                {owed.map((a) => (
                  <option key={a.email} value={a.email}>
                    {a.name}
                  </option>
                ))}
              </Form.Control>
            </Form.Group>
            <Form.Group controlId="exampleForm.ControlInput1">
              <Form.Label>Amount</Form.Label>
              <Form.Control type="number" placeholder={numeral(amount).format()} disabled />
            </Form.Group>
          </Form>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Close
          </Button>
          <Button variant="primary" onClick={() => settleUp(reciever, email, amount)}>
            Settle up
          </Button>
        </Modal.Footer>
      </Modal>
      <Row className="dashboard-layout-row">
        <VerticalNav groups={groups} />
        <Col xs={8} className="tiles">
          <Card bg="light" className="tiles">
            <Card.Header>
              <Row>
                <Col xs={10} data-testid="Dashboard">
                  Dashboard
                </Col>
                <Col>
                  <DropdownButton as={ButtonGroup} title="Settle up" id="bg-nested-dropdown">
                    <Dropdown.Item onClick={handleShow}>you owe</Dropdown.Item>
                    <Dropdown.Item onClick={handleShowOwed}>you owed</Dropdown.Item>
                  </DropdownButton>
                </Col>
              </Row>
            </Card.Header>
          </Card>

          <Row>
            <Col className="total-balance">
              <Card bg="light">
                <Card.Header>total Balance {numeral(totalBalance).format()}</Card.Header>
              </Card>
            </Col>
            <Col className="you-owe">
              <Card bg="light">
                <Card.Header>You owe {numeral(youOwe).format()}</Card.Header>
              </Card>
            </Col>
            <Col className="you-are-owed">
              <Card bg="light">
                <Card.Header>You are owed {numeral(youOwed).format()}</Card.Header>
              </Card>
            </Col>
          </Row>

          <Row className="owe-owed">
            <Col className="owe-column">
              <div className="you-owe-title">You Owe</div>
              <ListGroup>
                {owe.map((a) => (
                  <ListGroup.Item key={a.name} className="links-dashboard-groups">
                    name:{a.name}&nbsp;&nbsp;&nbsp;amt:{numeral(a.amount).format()}
                  </ListGroup.Item>
                ))}
              </ListGroup>
            </Col>
            <Col className="owed-column">
              <div className="you-owe-title">You Owed </div>
              <ListGroup>
                {owed.map((a) => (
                  <ListGroup.Item key={a.name} className="links-dashboard-groups">
                    name:{a.name}&nbsp;&nbsp;&nbsp;amt:{numeral(a.amount).format()}
                  </ListGroup.Item>
                ))}
              </ListGroup>
            </Col>
          </Row>
        </Col>
        <Col className="nav-dashboard">
          {invites.map((member) => (
            <ListGroup>
              <ListGroup.Item key={member} className="links-dashboard-groups">
                {member.invitedby} has invitted you to join group {member.groupName}
                &nbsp;
                <i
                  class="fas fa-check-circle accept"
                  onClick={() => accept(member.groupName, member.name)}
                ></i>
                &nbsp;
                <i class="fas fa-times-circle reject"></i>
              </ListGroup.Item>
            </ListGroup>
          ))}
        </Col>
      </Row>
    </div>
  );
}

export default Dashboard;
