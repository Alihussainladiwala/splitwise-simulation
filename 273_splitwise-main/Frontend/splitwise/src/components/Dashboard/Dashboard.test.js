/* eslint-disable react/jsx-filename-extension */
/* eslint-disable */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Dashboard from './Dashboard';
import { MemoryRouter } from 'react-router-dom'
import * as dashFunc from './Dashboard';


afterEach(cleanup);

it("renders", async()=>{
    const {asFragment} =  render(<Dashboard />, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot()
})

test('Check for header', async() => {
  const { getByTestId, getByText } = render(<Dashboard />, { wrapper: MemoryRouter });
  expect(getByTestId('Dashboard')).toHaveTextContent("Dashboard")
});
