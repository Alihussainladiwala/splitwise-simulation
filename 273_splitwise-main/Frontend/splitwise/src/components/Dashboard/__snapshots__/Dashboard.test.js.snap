// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="div-dashboard"
  >
    <div
      class="dashboard-layout-row row"
    >
      <div
        class="nav-dashboard"
      >
        <div
          class="nav-dashboard col"
        >
          <div
            class="flex-column nav"
          >
            <a
              class="button-dashboard-dashboard nav-link"
              href="#"
              role="button"
            >
              Dashboard
            </a>
            <a
              class="button-dashboard-recent-activity nav-link"
              href="#"
              role="button"
            >
              Recent Activity
            </a>
            <a
              class="button-dashboard-groups nav-link"
              href="#"
              role="button"
              variant="secondary"
            >
              Groups +
            </a>
            <div
              class="groups-list"
            >
              <input
                class="input-search"
                placeholder=". . . search"
                type="text"
              />
            </div>
            <a
              aria-disabled="true"
              class="nav-link disabled"
              data-rb-event-key="disabled"
              href="#"
              role="button"
              tabindex="-1"
            >
              Friends
            </a>
          </div>
        </div>
      </div>
      <div
        class="tiles col-8"
      >
        <div
          class="tiles card bg-light"
        >
          <div
            class="card-header"
          >
            <div
              class="row"
            >
              <div
                class="col-10"
                data-testid="Dashboard"
              >
                Dashboard
              </div>
              <div
                class="col"
              >
                <div
                  class="dropdown btn-group"
                  role="group"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="dropdown-toggle btn btn-primary"
                    id="bg-nested-dropdown"
                    type="button"
                  >
                    Settle up
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="total-balance col"
          >
            <div
              class="card bg-light"
            >
              <div
                class="card-header"
              >
                total Balance 0
              </div>
            </div>
          </div>
          <div
            class="you-owe col"
          >
            <div
              class="card bg-light"
            >
              <div
                class="card-header"
              >
                You owe 0
              </div>
            </div>
          </div>
          <div
            class="you-are-owed col"
          >
            <div
              class="card bg-light"
            >
              <div
                class="card-header"
              >
                You are owed 0
              </div>
            </div>
          </div>
        </div>
        <div
          class="owe-owed row"
        >
          <div
            class="owe-column col"
          >
            <div
              class="you-owe-title"
            >
              You Owe
            </div>
            <div
              class="list-group"
            />
          </div>
          <div
            class="owed-column col"
          >
            <div
              class="you-owe-title"
            >
              You Owed 
            </div>
            <div
              class="list-group"
            />
          </div>
        </div>
      </div>
      <div
        class="nav-dashboard col"
      />
    </div>
  </div>
</DocumentFragment>
`;
